assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+const baseUrl = 'https://pokeapi.co/api/v2/';
+
+let pkmnList = []; // full pkmn list
+let pkmnTypes = []; // list of types
+let typesCorespondingPkmn = []; // List of pkmn that have each type in {TYPE: NAME, LIST: [PKMN, PKMN]} format
+let moves = []; // list of moves
+let currentMoves = [];
+let currentStats = [];
+let builtPkmn = []; // pkmn that have been generated by user already, onclick event listener will check this list first... eventually.
+let pkmnListForTypeFilter = []
+let currentPkmn;
+let types = ['Bug', 'Dark', 'Dragon', 'Electric', 'Fairy', 'Fighting',
+    'Fire', 'Flying', 'Ghost', 'Grass', 'Ground', 'Ice', 'Normal',
+    'Poison', 'Psychic', 'Rock', 'Steel', 'Water'
+];
+
+
+let pkmnListDiv = document.getElementById("pkmnList");
+let pkmnPic = document.getElementById("terminalScreen");
+let pkmnStats = document.getElementById("statSection");
+let moveset = document.getElementById("scrollingMoveset");
+let type1 = document.getElementById("leftType").getElementsByTagName("p")[0];
+let type2 = document.getElementById("rightType").getElementsByTagName("p")[0];
+let idBox = document.getElementById("pkmnIdBox");
+let downId = document.getElementById("downId");
+let upId = document.getElementById("upId");
+let pkmnName = document.getElementById("pkmnName");
+let terminalButton = document.getElementById("terminalButton");
+let typesDiv = document.getElementById("types");
+let clearFilter = document.getElementById("clearFilter");
```

Comment:
Good thinking on trying to `cache` some of these off so you dont have to keep fetching them.